homeassistant/components/weheat/binary_sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Platform for binary sensor integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from weheat.abstractions.heat_pump import HeatPump
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import WeheatConfigEntry
+from .coordinator import WeheatDataUpdateCoordinator
+from .entity import WeheatEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class WeHeatBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Weheat binary sensor entity."""
+
+    value_fn: Callable[[HeatPump], StateType]
+
+
+BINARY_SENSORS = [
+    WeHeatBinarySensorEntityDescription(
+        translation_key="indoor_unit_water_pump_state",
+        key="indoor_unit_water_pump_state",
+        value_fn=lambda status: status.indoor_unit_water_pump_state,
+    ),
+    WeHeatBinarySensorEntityDescription(
+        translation_key="indoor_unit_auxiliary_pump_state",
+        key="indoor_unit_auxiliary_pump_state",
+        value_fn=lambda status: status.indoor_unit_auxiliary_pump_state,
+    ),
+    WeHeatBinarySensorEntityDescription(
+        translation_key="indoor_unit_dhw_valve_or_pump_state",
+        key="indoor_unit_dhw_valve_or_pump_state",
+        value_fn=lambda status: status.indoor_unit_dhw_valve_or_pump_state,
+    ),
+    WeHeatBinarySensorEntityDescription(
+        translation_key="indoor_unit_gas_boiler_state",
+        key="indoor_unit_gas_boiler_state",
+        value_fn=lambda status: status.indoor_unit_gas_boiler_state,
+    ),
+    WeHeatBinarySensorEntityDescription(
+        translation_key="indoor_unit_electric_heater_state",
+        key="indoor_unit_electric_heater_state",
+        value_fn=lambda status: status.indoor_unit_electric_heater_state,
```

Comment:
Would it make sense to give these the device class running and remove `state` from the names?

This would make the state show `Running` and `Not running`

Reply:
Ehm, maybe for the pumps. Not sure about the last 2

Reply:
I guess we could just remove State from the name, not sure if 'running' makes more sense? 
Should I remove state from the name, or als from the keys?

Reply:
Changed everything accept the boiler state to running, because the boiler is only an indication of the CV boiler is allowed to be used.